src/Datadog.Trace/Util/StreamHelper.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.IO;
+
+namespace Datadog.Trace.Util
+{
+    internal static class StreamHelper
+    {
+        /// <summary>
+        /// Gets underlyding buffer directly when using supported framework version,
+        /// or allocates new buffer if not supported.
+        /// </summary>
+        /// <param name="byteStream">Stream's buffer to extract</param>
+        /// <returns>Underlying or new buffer</returns>
+        public static ArraySegment<byte> GetMemoryBuffer(this MemoryStream byteStream)
+        {
+            ArraySegment<byte> buffer;
+
+#if !NET45
```

Comment:
nit: My preference for readability would be to have separate `#if !NET45` and `#else` blocks to describe the entire assignment logic instead of breaking up the operation like it is here. It's not critical but I wanted to provide that feedback 😄

Reply:
Actually I really thought that way also. Why I decided to go the other route was because it wasn't that much more complex, It's hidden in the helper and mainly doesn't use line duplication. In Visual Studio inactive framework code segments are grayed out, so I thought going without line duplication will be safer as the other duplication might be invisible.